pkg/utils/validator/clusterresourceoverride.go

```diff
@@ -0,0 +1,99 @@
+package validator
+
+import (
+	"context"
+	"fmt"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	fleetv1alpha1 "go.goms.io/fleet/apis/placement/v1alpha1"
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+)
+
+// ValidateClusterResourceOverride validates cluster resource override fields and returns error.
+func ValidateClusterResourceOverride(ctx context.Context, client client.Client, operation admissionv1.Operation, cro fleetv1alpha1.ClusterResourceOverride) error {
+	allErr := make([]error, 0)
+
+	// Check if the resource is being selected by resource name
+	if err := validateResourceSelected(cro); err != nil {
+		return err
+	}
+	// List of cluster resource overrides
+	croList, err := listClusterResourceOverrideList(ctx, client)
+	if err != nil {
+		return err
+	}
+	// Check if the override count limit has been reached
+	if !ValidateClusterResourceOverrideLimit(operation, croList) {
+		allErr = append(allErr, fmt.Errorf("at most 100 cluster resource overrides can be created"))
+	}
+
+	// Check if the override count limit for the resources has been reached
+	if err = ValidateClusterResourceOverrideResourceLimit(operation, cro, croList); err != nil {
+		allErr = append(allErr, err)
+	}
```

Comment:
as discussed, let's treat these checks as a special validation for CREATE/UPDATE and ResourceLimit is expensive, so skip for other operations too.

Please refer to CRP validation webhook.